dask/dataframe/optimize.py

```diff
@@ -19,7 +20,11 @@
         # Merge column access into bcolz loading
         RewriteRule((getitem, (dataframe_from_ctable, a, b, c, d), e),
                     (dataframe_from_ctable, a, b, e, d),
-                    (a, b, c, d, e)))
+                    (a, b, c, d, e)),
+        RewriteRule((getitem, (Castra.load_partition, '~c', '~part', '~cols1'),
```

Comment:
Adding this optimization, without ever inserting castra into dask ourselves makes this seem like an optimization that should only exist if castra is on the system (only add this rule if castra is importable?). Unless you plan to use castra in other places, and this is just the first one.